[AMORO-2200] Optimize the docker script for amoro

tcodehuber opened the pull request:
<!--
Thanks for sending a pull request!

Here are some tips for you:
  1. If this is your first time, please read our contributor guidelines: https://amoro.netease.com/how-to-contribute/
  2. If the PR is related to an issue in https://github.com/NetEase/amoro/issues, add '[AMORO-XXXX]' in your PR title, e.g., '[AMORO-XXXX] Your PR title ...'.
  3. If the PR is unfinished, add '[WIP]' in your PR title, e.g., '[WIP][AMORO-XXXX] Your PR title ...'.
-->

## Why are the changes needed?
<!--
Please clarify why the changes are needed. For instance,
  1. If you add a feature, you can talk about its use case.
  2. If you fix a bug, you can clarify why it is a bug.
  3. Use Fix/Resolve/Close #{ISSUE_NUMBER} to link this PR to its related issue
-->

fix:#2200.

## Brief change log
<!--
Clearly describe the changes made in modules, classes, methods, etc.
-->

-

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

baiyangtx commented on docker/quickdemo/Dockerfile:
```diff
@@ -25,7 +25,7 @@
 ARG AMORO_TAG=master-snapshot
 
-FROM eclipse-temurin:8-jdk-jammy as builder
+FROM alpine as builder
```
There is a conflict between alpine and Flink, which causes Flink to fail to start the task manager.

baiyangtx reviewed (commented):
I noticed that this PR has done more than one thing. 
1. Modified the Docker image. 
2. Implemented the Spark optimizer. 
These two things are relatively independent, please submit separate PRs to complete them, which is important for maintaining a clear commit history.

baiyangtx commented:
I noticed that this PR provides a Spark optimizer job but does not provide the implementation related to the Spark optimizer container. 
Can you implement this part of the code? Otherwise, we can only use the Spark optimizer in an external container.

AND, 

You can refer to the Flink optimizer module, and the Spark optimizer module should be implemented in a separate module. This way, the project can be compiled more flexibly.

tcodehuber commented:
> I noticed that this PR has done more than one thing.
> 
> 1. Modified the Docker image.
> 2. Implemented the Spark optimizer.
> 
> These two things are relatively independent, please submit separate PRs to complete them, which is important for maintaining a clear commit history.

sorry, my fault. Now i have correct it.

baiyangtx reviewed (commented):
Have you tested the quickdemo process? As far as I know, there is a conflict between the Alpine image and the Flink engine, which prevents the Flink task from starting.

tcodehuber commented:
> Have you tested the quickdemo process? As far as I know, there is a conflict between the Alpine image and the Flink engine, which prevents the Flink task from starting.

Not yet. How can i close this PR?

baiyangtx commented:
> > Have you tested the quickdemo process? As far as I know, there is a conflict between the Alpine image and the Flink engine, which prevents the Flink task from starting.
> 
> Not yet. How can i close this PR?

I will close it.

baiyangtx commented:
The Spark optimizer is useful. Are you willing to submit a new Pull Request (PR)?  @tcodehuber

tcodehuber commented:
> The Spark optimizer is useful. Are you willing to submit a new Pull Request (PR)? @tcodehuber

OK, I will try it later.
